### DISCUSSION TITLE
对于saucenao多个图库的结果，bot是自己有一个优先级？

### DISCUSSION THREAD

**Discussion by Miuzarte** [Tsuk1ko/CQ-picfinder-robot#199] [一般讨论]
pixiv的结果相似度更高，但不加参数默认得到mangadex

<details>

![652bcaa15758f05c.png](https://user-images.githubusercontent.com/66856838/119966038-85628200-bf70-11eb-9673-020de7a25851.png)
</details>

**Comment by Tsuk1ko** [comment]
发一下原图，我回去试一下

**Reply by Miuzarte** [reply]
<details>

![e2ea8b44533f1c2.jpg](https://user-images.githubusercontent.com/66856838/119968922-8fd24b00-bf73-11eb-9686-8a6b533e6252.jpg)
</details>

**Comment by Tsuk1ko** [comment]
试了下发现 saucenao 返回的内容不会按相似度排序，我一直是取第一个结果的，现在我人为排序了下

**Reply by Tsuk1ko** [reply]
但这样某些搜图 pixiv 的相似度会比 danbooru 低……我还是得人为加个权重

**Reply by Miuzarte** [reply]
> 但这样某些搜图 pixiv 的相似度会比 danbooru 低……我还是得人为加个权重

反正有合并发送,不如~~我全都要!~~

**Reply by Tsuk1ko** [reply]
（那不行

**Reply by Tsuk1ko** [reply]
不行，不能完全按相似度搞，我发现 saucenao 的排序大部分情况下其实挺合理的，全按相似度反而不太科学，我还是只针对首位不是 pixiv 的情况再从结果里找相似度更高的 pixiv 比较好

**Reply by Miuzarte** [reply]
加开关!

```jsonc
//结果中总是展示以下的图库
pixiv: ""
danbooru: ""
-----------------
alwaysshow: [pixiv,danbooru]
```

**Reply by Tsuk1ko** [reply]
不加！太麻烦！
